src/Microsoft.Health.SqlServer/Features/Client/RetrySqlCommandWrapperFactory.cs

```diff
@@ -14,21 +13,20 @@ namespace Microsoft.Health.SqlServer.Features.Client
     /// </summary>
     internal class RetrySqlCommandWrapperFactory : SqlCommandWrapperFactory
     {
-        private readonly IAsyncPolicy _retryPolicy;
+        private readonly SqlRetryLogicBaseProvider _sqlRetryLogicBaseProvider;
 
-        public RetrySqlCommandWrapperFactory(ISqlServerTransientFaultRetryPolicyFactory sqlTransientFaultRetryPolicyFactory)
+        public RetrySqlCommandWrapperFactory(SqlRetryLogicBaseProvider sqlRetryLogicBaseProvider)
         {
-            EnsureArg.IsNotNull(sqlTransientFaultRetryPolicyFactory, nameof(sqlTransientFaultRetryPolicyFactory));
-
-            _retryPolicy = sqlTransientFaultRetryPolicyFactory.Create();
+            EnsureArg.IsNotNull(sqlRetryLogicBaseProvider, nameof(sqlRetryLogicBaseProvider));
+            _sqlRetryLogicBaseProvider = sqlRetryLogicBaseProvider;
         }
 
         /// <inheritdoc/>
         public override SqlCommandWrapper Create(SqlCommand sqlCommand)
         {
             return new RetrySqlCommandWrapper(
                 base.Create(sqlCommand),
```

Comment:
In the future, we'd like users to invoke `SqlConnection.OpenAsync` before they execute their commands, but do we need to include this inside of the `Execute*Async*` methods for `RetrySqlCommandWrapper` now that we can rely on the framework to handle re-opening the connection?

If not, we can probably remove the logic to open the connection explicitly again in the `RetrySqlCommandWrapperFactory` and perhaps even just delete `RetrySqlCommandWraper`. We can instead change this factory to do something like:

```csharp
sqlCommand.RetryLogicProvider = _sqlRetryLogicBaseProvider;
return new SqlCommandWrapper(base.Create(sqlCommand));
```